### ISSUE TITLE
Collection / navigation properties are causing ModelState validation issues since upgrading to 5.2.1

### ISSUE THREAD

**Issue by jm-crimsontide** [linq2db/linq2db#4197]
### Describe your issue

Since upgrading to 5.2.1 (from 3.7.0), we've hit an issue where ModelState validation (within a C# WebAPI) now fails on non-nullable navigation properties.

Both 3.7.0 and 5.2.1 appear to define the same collection / navigation property such as the following, which is not-nullable:

```csharp
[Association(ThisKey = nameof(AuditID), OtherKey = nameof(Order.AuditID))]
public IEnumerable<Order> Orders { get; set; } = null!;
```

However the 5.2.1 version now fails ModelState validation, as the field now appears to be required:
![image](https://github.com/linq2db/linq2db/assets/138590451/89571009-0282-475c-addf-9215c37ca885)

ModelState was not failing with 3.7.0

This makes sense to an extent, as the property should now be marked as nullable (see the `?` added):

```csharp
[Association(ThisKey = nameof(AuditID), OtherKey = nameof(Order.AuditID))]
public IEnumerable<Order>? Orders { get; set; } = null!;
```

But it's not clear how to either a) get the nullability added in as part of the model generation (using JSON config and an interceptor) or b) how to toggle the ModelState to behave the same way as 3.7.0. I have tried using the following at application startup, however it doesn't appear to work:

```csharp
Configuration.UseNullableTypesMetadata = true;
```

### Steps to reproduce

I'm unable to share our private code however will try to summarize the key items:

1. Create a new C# WebAPI using .NET 6, along with the service and repository classes needed
2. Implement linq2db and generate a model from a target database
3. Run the application
4. Use Postman to POST to an endpoint, but do not pass in the `Orders` property
5. ModelState validation will fail as `Orders` has not been passed in

I can share some of the details of the JSON configuration if that helps (connection string removed and path to the .dll changed):

```json
{
  "general": {
    "provider": "SQLServer",
    "connection": "",
    "output": "./",
    "overwrite": true,
    "customize": "interceptors.dll"
  },
  "schema": {
    "objects": [ "table", "foreign-key" ]
  },
  "dataModel": {
    "association-name": {
      "case": "t4"
    },
    "association-single-backreference-name": {
      "case": "t4"
    },
    "entity-class-name": {
      "case": "t4",
      "ignore_all_caps": true,
      "pluralization": "none"
    },
    "entity-column-property-name": {
      "case": "t4",
      "ignore_all_caps": true
    },
    "partial-entities": true
  }
}
```

### Environment details

`Linq To DB` version: 5.2.1

Database (with version): Azure SQL Server

Operating system: Windows 10 Enterprise

.NET Version: 6

**Comment by jm-crimsontide** [comment]
Quick update: having spent days looking into this issue, may have found a suitable workaround. Just testing changes before updating this thread!

**Comment by jm-crimsontide** [comment]
The issue appears to be resolved by using the following value in the JSON configuration file:

```json
  "dataModel": {
    "association-collection": "System.Collections.Generic.IEnumerable<>?",
    "association-name": {
      "case": "t4"
    },
```

**Comment by MaceWindu** [comment]
To disable NRT you can use code generation option:
```json
{
    ....
    "code": {
        "nrt": false
    }
}
```


Personally I think proper approach is to exclude associations from model binding using [`BindNever`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.modelbinding.bindneverattribute?view=aspnetcore-5.0) attribute. It could be done using custom interceptor. Check https://linq2db.github.io/articles/CLI.html#data-model-interceptors 

Basically you need to implement `PreprocessAssociation` interceptor and add attribute to `associationModel.[Property/BackreferenceProperty].CustomAttributes`
